source/_integrations/tesla_fleet.markdown

```diff
@@ -40,30 +40,140 @@ The Tesla Fleet API {% term integration %} exposes various sensors from Tesla ve
+- A [Tesla](https://tesla.com) account
+- A [Developer Application](https://developer.tesla.com/en_US/dashboard)
+- A web domain and host that you can serve your public key file from. Some free options include:
+  - [FleetKey.cc](https://fleetkey.cc)
+  - [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/WebsiteHosting.html)
+  - [Cloudflare Pages](https://pages.cloudflare.com/)
+  - [Firebase Hosting](https://firebase.google.com/docs/hosting)
+  
 
 {% include integrations/config_flow.md %}
 
+{% details "Hosting a Public/Private Key Pair" %}
+
+While the [Tesla Fleet API documentation Step 3](https://developer.tesla.com/docs/fleet-api/getting-started/what-is-fleet-api#step-3-generate-a-public-private-key-pair) mentions this as a later step, it is recommended that you do this first to ensure key reachability before the rest of the integration.
+
+1. Open a new terminal on your computer.
+2. Run this command to create a private key: `openssl ecparam -name prime256v1 -genkey -noout -out private-key.pem`
+3. Run this command to create and associate a public key with it: `openssl ec -in private-key.pem -pubout -out public-key.pem`
+4. Rename the public key file to `com.tesla.3p.public-key.pem`. This needs to be exactly this for Tesla's API to partner with your account correctly.
+5. Backup both these files somewhere safe and private for access later.
+6. Upload the public key file to your domain at the path `/.well-known/appspecific/com.tesla.3p.public-key.pem`. For example, if your domain is `https://my.domain.com`, the public key file must be accessible at `https://my.domain.com/.well-known/appspecific/com.tesla.3p.public-key.pem` . Do not use redirection logic to handle this, or the Tesla API will not recognize your app later in the process.
+
+{% enddetails %}
+
+{% details "Setting up the Developer Application" %}
+
+These steps are also summarized in the [Tesla Fleet API documentation Step 2](https://developer.tesla.com/docs/fleet-api/getting-started/what-is-fleet-api#step-2-create-an-application), but it is recommended to follow the steps below to cover specific details.
+
+1. If you have not already, set up your [Tesla Developer account](https://developer.tesla.com/teslaaccount). Confirm that you have a verified email and multi-factor authentication set up.
+2. Navigate to the [Developer dashboard](https://developer.tesla.com/en_US/dashboard). Change your locale if needed to wherever your account is based, using the globe icon at the top-right corner.
+3. Click '**Create New Application**'. This should launch a new page with the header '**Create Fleet API Application**'.
```

Comment:
```suggestion
3. Select **Create New Application**. This should launch a new page with the header **Create Fleet API Application**.
```
We favor [select over click](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click)